src/plugins/data/common/opensearch_query/filters/geo_shape_filter.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GeoShapeRelation } from '@opensearch-project/opensearch/api/types';
+import { Filter, FilterMeta } from './meta_filter';
+
+export type Position = number[];
+
+export interface PreIndexedShapeFilter {
+  index: string;
+  id: string;
+  path: string;
+  routing?: string;
+}
+
+export interface Polygon {
+  type: 'Polygon';
+  coordinates: Position[][];
+}
+
+export interface MultiPolygon {
+  type: 'MultiPolygon';
+  coordinates: Position[][][];
+}
+
+// TODO: support other geometries too.
+export type ShapeFilter = Polygon | MultiPolygon;
+
+export type GeoShapeFilterMeta = FilterMeta & {
+  params: {
+    shape?: ShapeFilter;
+    indexed_shape?: PreIndexedShapeFilter;
+    relation?: GeoShapeRelation;
+  };
+};
+
+export type GeoShapeFilter = Filter & {
+  meta: GeoShapeFilterMeta;
+  geo_shape: any;
+};
+
+export const isGeoShapeFilter = (filter: any): filter is GeoShapeFilter => filter?.geo_shape;
+
+export const getGeoShapeFilterField = (filter: GeoShapeFilter) => {
+  return (
```

Comment:
NIT: should we explicitly define `return undefined` to make it more clearer?

```
export const getGeoShapeFilterField = (filter: GeoShapeFilter) => {
  if (filter.geo_shape === undefined) {
    return undefined;
  }
  return Object.keys(filter.geo_shape).find((key) => key !== 'ignore_unmapped');
};
```

Reply:
Updated